flow/ApiVersion.h.cmake

```diff
@@ -0,0 +1,72 @@
+/*
+ * ProtocolVersion.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "flow/Trace.h"
+#include <cstdint>
+
+constexpr int noBackwardsCompatibility = 13;
+
+// The first check second expression version doesn't need to change because it's just for earlier protocol versions.
+#define API_VERSION_FEATURE(v, x)                                                                                      \
+	static_assert(v <= @FDB_AV_LATEST_VERSION@, "Feature protocol version too large");                                   \
+	struct x {                                                                                                         \
+		static constexpr uint64_t apiVersion = v;                                                                      \
+	};                                                                                                                 \
+	constexpr bool has##x() const { return this->version() >= x ::apiVersion; }                                        \
+	static constexpr ApiVersion with##x() { return ApiVersion(x ::apiVersion); }
+
+class ApiVersion {
+	int _version;
+
+public:
+	// Statics.
+	constexpr static int LATEST_VERSION = @FDB_AV_LATEST_VERSION@;
+	constexpr static int FUTURE_VERSION = @FDB_AV_FUTURE_VERSION@;
```

Comment:
I don't think we have any uses of FUTURE_VERSION, it wouldn't pass the isValid() check either. Probably better remove it